inc/commondbtm.class.php

```diff
@@ -271,20 +271,43 @@ public static function getFromIter(DBmysqlIterator $iter) {
     * @return boolean|array
     */
    public function getFromDBByCrit(array $crit) {
+      return $this->getFromDBByRequest(['WHERE' => $crit]);
+   }
+
+   /**
+    * Retrieve an item from the database by request. The request is an array
+    * similar to the one expected in DB::request().
+    *
+    * @since 9.3
+    *
+    * @see DB::request()
+    *
+    * @param array $request expression
+    *
+    * @return boolean true if succeed else false
+    **/
+   public function getFromDBByRequest(array $request) {
       global $DB;
 
-      $crit = ['SELECT' => 'id',
-               'FROM'   => $this->getTable(),
-               'WHERE'  => $crit];
+      // Limit the request to the useful expressions
+      $request = array_intersect_key($request, [
+         'WHERE' => '',
+         'LEFT JOIN' => '',
+         'INNER_JOIN' => '',
+         'START' => '',
+         'LIMIT' => '',
+      ]);
+      $request['FROM'] = $this->getTable();
+      $request['SELECT'] = $this->getTable() . '.id';
 
-      $iter = $DB->request($crit);
+      $iter = $DB->request($request);
       if (count($iter) == 1) {
          $row = $iter->next();
          return $this->getFromDB($row['id']);
```

Comment:
Not directly related to the current pull request; but we should avoid calling `getFromDB` here, since it will make an extra call to the database. Using `getFromIter` sounds a better solution.

Reply:
getFromIter itseld calls getFromDB, then this is not better than the current solution.

Also, getFromDB() calls post_getFromDB(), then I think we may hydrate the isntance with the full row, then call post_getFromDB().